### PR TITLE
feat(rnd): Add support for supabase locally

### PR DESCRIPTION
**Pull request by aarushik93**
### Background

This PR adds supabase to local dev. Allowing all users to have an authentication service running locally. 
As supabase comes with some heafty docker compose configuration, I have added the repo as a submodule. 
On top of that I have created a combined docker compose that takes our base autogpt service docker compse and supabase docker compose and extends the two and bridges the two networks so supabase and agpt can talk to each other.

I also had a lot of issues with preflight cors failures when connecting the frontend to kong (api gateway). It appears to be a known issue and different ppl have tried various workarounds. Nothing I tried worked: 
- Disabling CORS on my browser
- Setting up Nginx to proxy the reequest

However, running the frontend from _outside_ of docker seems to work.

### Changes 🏗️

Submoduled supabase
Created combined docker compose
Update environment variables to use local supabase
Commented out frontend in docker compose

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [ ] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

File: rnd/docker-compose.yml [hunk 1]
```diff
@@ -161,41 +167,41 @@ services:
...
-      - NEXT_PUBLIC_AGPT_WS_SERVER_URL=ws://localhost:8001/ws
-      - NEXT_PUBLIC_AGPT_MARKETPLACE_URL=http://localhost:8015/api/v1/market
     ports:
-      - "3000:3000"
+      - "8015:8015"
     networks:
       - app-network
 
+#  frontend:
+#    build:
```
Comment (aarushik93) [inline rnd/docker-compose.yml:180]: Leaving it commented out as I do want to figure out later on why this doesn't work

File: rnd/docker-compose.yml [hunk 2]
```diff
@@ -72,15 +69,18 @@ services:
...
       - REDIS_HOST=redis
       - REDIS_PORT=6379
       - REDIS_PASSWORD=password
-      - AUTH_ENABLED=false
+      - ENABLE_AUTH=true
       - PYRO_HOST=0.0.0.0
       - EXECUTIONMANAGER_HOST=executor
     ports:
-      - "8000:8000"
+      - "8006:8000"
```
Comment (aarushik93) [inline rnd/docker-compose.yml:83]: kong runs on 8000 so exposing 8006 instead

File: rnd/docker-compose.yml [hunk 3]
```diff
@@ -11,21 +11,18 @@ services:
       timeout: 5s
       retries: 5
     ports:
-      - "5432:5432"
+      - "5433:5432"
```
Comment (aarushik93) [inline rnd/docker-compose.yml:14]: supabase's pg runs on 5432, so exposing 5433

File: rnd/docker-compose.yml (hunk 1 above)
Comment (majdyz) [reply rnd/docker-compose.yml:180]: so prod doesn't even use this docker-compose ?

**Review by majdyz — APPROVED** [review]

File: rnd/docker-compose.yml [hunk 4]
```diff
@@ -11,21 +11,18 @@ services:
...
     networks:
       - app-network
-    
+
   migrate:
-    build:
-      context: ../
-      dockerfile: rnd/autogpt_server/Dockerfile
-      target: server
+    image: autogpt_server:latest
```
Comment (Swiftyos) [inline rnd/docker-compose.yml:19]: This is what is causing the docker error

**Pull request merged**